tutorial/guias/Paso_a_paso_DIA_6.md

```diff
@@ -0,0 +1,375 @@
+        else:
+            self.snake.trim()
+
+        Renderer.render_game_state(
+            self.snake.body,
+            self.fruit.position,
+            self.game_window,
+            Color.GREEN.value,
+            Color.WHITE.value,
+            Color.BLACK.value,
+        )
+
+        if self.game_over():
+            Renderer.render_game_over(
+                self.score,
+                self.game_window,
+                self.window_x,
+                self.window_y,
+                Color.RED.value,
+                "times new roman",
+                50,
+            )
+            time.sleep(2)
+            pygame.quit()
+            sys.exit()
+
+        Renderer.render_score(
+            self.score, self.game_window, Color.WHITE.value, "times new roman", 20
+        )
+        pygame.display.update()
+        self.fps.tick(self.snake_speed)
+```
+
+Todo nuestro juego es un gran ciclo que no acaba hasta que el jugador pierde o cierra voluntariamente el juego. Es por eso que nuestra función es un gran ciclo `while` que no tiene condición de parada (`while True:`). Así, nuestro juego se mantedrá ejecutándose mientras las condiciones para terminarlo no se cumplan.
+
+En primer lugar, verificamos si el juego está pausado, en cuyo caso no hacemos más nada en esta iteración y dejamos que el ciclo `while` se vuelva a ejecutar. Luego actualizamos la dirección de la serpiente. Si no se modifica con las flechas del teclado, entonces se mantiene la misma dirección. Con esta dirección movemos a la serpiente usando las funciones `change_position` y `move` de la clase `Snake`. Recordemos que la función `move` provoca que la serpiente crezca.
+
+Lo siguiente es verificar si la serpiente comió una fruta. Esto sucede cuando la posición de la fruta y la de la cabeza de la serpiente coinciden. En este caso, aumentamos el score en 10, generamos una nueva fruta e incrementamos la velocidad del juego. Como ya la función `move` hizo crecer a la serpiente, no nos tenemos que preocupar por ello.
+
+En caso de que no se haya comido una fruta, entonces tenemos que revertir el crecimiento de la serpiente y para eso usamos la función `trim`.
+
+Luego usamos la función `render_game_sate` de la clase `Renderer` para representar todos los elementos del juego. Después toca verificar si hemos perdido, o sea, si la función `game_over` nos devuelve `True` en las condiciones actuales del juego. En caso afirmativo, procedemos a dibujar la pantalla de *game over*. Usamos la fuente *time new roman* con tamaño 50. Dejamos esta pantalla durante 2 segundos con `time.sleep(2)` y luego cerramos el juego.
+
+Si no hemos perdido aún, representamos nuestro score en pantalla. Usaremos para ello la fuente *times new roman* pero esta vez con tamaño 20.
+
+Usamos la función `pygame.display.update()` para que nuestra pantalla se actualice con la situación actual del juego y adaptamos los *fraps per second* a la velocidad del juego con `self.fps.tick(self.snake_speed)`.
```

Comment:
Again, I think it is `frames per second`, but now I'm starting to think it was not a typo on your side 😂. So, maybe you are right.